### PR TITLE
Increase torch compatibility

### PR DESCRIPTION
**Pull request by boeddeker**
- I added some properties and methods to ComplexTensor and forwarded them to the rel and imag part.
 - torch supports `dim` and `axis`. To be a drop in replacement, I changed the signature of some functions to have `*args` and `**kwargs` (Sadly torch has multiple signatures for most functions).
 - I also changed the repr of ComplexTensor. I think with indents it is better to read.

### CODE REVIEW

File: torch_complex/tensor.py [hunk 1]
```diff
@@ -532,9 +535,17 @@ def new_tensor(
     def numpy(self) -> numpy.ndarray:
         return self.real.numpy() + 1j * self.imag.numpy()
 
+    def __array__(self):
+        # https://numpy.org/devdocs/user/basics.dispatch.html
```
Comment (kamo-naoyuki) [inline torch_complex/tensor.py:539]: Great! I don't know this way. Thanks.

File: torch_complex/functional.py [hunk 2]
```diff
@@ -46,6 +46,9 @@ def einsum(equation, operands):
     >>> numpy.testing.assert_allclose(test.numpy(), valid)
 
     """
+    if len(operands) == 1 and isinstance(operands, (tuple, list)):
```
Comment (kamo-naoyuki) [inline torch_complex/functional.py:49]:
```suggestion
    if len(operands) == 1:
```

`operands` is always tuple if * is added.

File: torch_complex/functional.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def einsum(equation, *operands):
     >>> numpy.testing.assert_allclose(test.numpy(), valid)
 
     """
-    if len(operands) == 1 and isinstance(operands, (tuple, list)):
+    if len(operands) == 1:
```
Comment (boeddeker) [inline torch_complex/functional.py:49]: Sorry, I didn't check your modification.

Einsum also supports a single argument.

The second part in my if statement was wrong.
It should check it the entry in the tuple is a tuple or list (Not if the tuple is a tuple or list).
i.e. `if len(operands) == 1 and isinstance(operands[0], (tuple, list)):` (I forgot the `[0]`)

The reason, why this is necessary, is that torch took the same function name as numpy, but initially they had a different signature.
At some point (Not sure, when) they allowed also the numpy style call.

Comment (kamo-naoyuki) [reply torch_complex/functional.py:49]: I see. Could you fix again?

Comment (boeddeker) [reply torch_complex/functional.py:49]: Sure, I will make a new PR.

**Pull request merged**